app/lib/katello/resources/discovery/container_discovery.rb

```diff
@@ -0,0 +1,123 @@
+module Katello
+  class ContainerDiscovery < RepoDiscovery
```

Comment:
I think for the autoloader to work this needs to be `Katello::Resources::Discovery::ContainerDiscovery`. Then given the repetition I'd suggest to rename this to `Katello::Resources::Discovery::Container`.

Reply:
Aah..Ya..That works..The earlier manual loading was cause the file wasn't getting loaded..Removed that and renamed this..